src/migration/1693322911755-migration.ts

```diff
@@ -0,0 +1,247 @@
+		   RETURN OLD;
+	   END
+	   $function$
+	   ;`);
+        await queryRunner.query(`CREATE OR REPLACE FUNCTION ${schemaName}.log_create()
+		 RETURNS trigger
+		 LANGUAGE plpgsql
+		AS $function$ 
+			   BEGIN
+				   IF OLD.pin IS NULL
+				   THEN
+					   INSERT INTO ${schemaName}.pin_audit_log 
+					   (pin,pids,title_number,land_title_district,from_title_number,from_land_title_district,title_status,live_pin_id,pin_created_at,action) 
+					   VALUES (NEW.pin,NEW.pids,NEW.title_number,NEW.land_title_district,NEW.from_title_number,NEW.from_land_title_district,NEW.title_status,NEW.live_pin_id,now(),'C');
+				   ELSE
+				   INSERT INTO ${schemaName}.pin_audit_log 
+				   (pin,pids,title_number,land_title_district,from_title_number,from_land_title_district,title_status,live_pin_id,ACTION,expired_at) 
+				   VALUES (OLD.pin,OLD.pids,OLD.title_number,OLD.land_title_district,OLD.from_title_number,OLD.from_land_title_district,OLD.title_status,OLD.live_pin_id,'R',now());
+				   INSERT INTO ${schemaName}.pin_audit_log 
+					   (pin,pids,title_number,land_title_district,from_title_number,from_land_title_district,title_status,live_pin_id,pin_created_at,action) 
+					   VALUES (NEW.pin,NEW.pids,NEW.title_number,NEW.land_title_district,NEW.from_title_number,NEW.from_land_title_district,NEW.title_status,NEW.live_pin_id,now(),'C');
+				   END IF;
+				   
+				   RETURN NEW;
+			   END
+			   $function$
+		;`);
+
+        // Parcel pin table (may not exist on server?)
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.parcel_pin DROP CONSTRAINT parcel_pin_live_pin_id_fkey;`,
+        );
+        await queryRunner.query(
+            `DROP TABLE IF EXISTS ${schemaName}.parcel_pin;`,
+        );
+
+        // Active pin table
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.active_pin ALTER COLUMN pid TYPE varchar(500) USING pid::varchar(500);`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.active_pin RENAME COLUMN pid TO pids;`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.active_pin DROP COLUMN IF EXISTS parcel_status;`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.active_pin ALTER COLUMN address_line_1 DROP NOT NULL;`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.active_pin ALTER COLUMN city DROP NOT NULL;`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.active_pin RENAME COLUMN province TO province_abbreviation;`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.active_pin RENAME COLUMN other_geographic_division TO province_long;`,
```

Comment:
I think we don't need to include province_long in the table (sorry not sure if this is removed later)

Reply:
Do we need to store this anyway for non-Canada / USA addresses?